### ISSUE TITLE
Better reporting for AngularParserException{NgParserWarningCode.INVALID_MICRO_EXPRESSION}

### ISSUE THREAD

**Issue by tonosama-atlacatl** [dart-lang/angular#1739]
Request to get a more meaningful description of what generated:
AngularParserException{NgParserWarningCode.INVALID_MICRO_EXPRESSION}

Environment:

- Dart SDK: Dart VM version: 2.2.0 (Tue Feb 26 15:04:32 2019 +0100) on "macos_x64"
- AngularDart Version: 5.2.0
- OS: MacOS Mojave 10.14.3
- angular_components: 0.12.0

I have two projects that I'm trying to migrate to SDK 2 and Angular 5. 
Project **A** contains my components mixed in with some angular_components.
Project **B** depends on Project A, a simple webapp.

When I compile Project A, build_runner claims a success, and everything is A-OK.
When I compile Project B, I get all these "SEVERE violations" that point to a .dart file, but no mention of what specific line/code issued the violation. I get cryptic stuff like this:

[SEVERE] angular on package:Project_A/src/components/common/action_bar/action_bar.dart (cached):
Unhandled exception in the AngularDart compiler!

Please report a bug: https://github.com/dart-lang/angular/issues/new
AngularParserException{NgParserWarningCode.INVALID_MICRO_EXPRESSION}
package:angular_ast/src/expression/micro/scanner.dart 159:5                     NgMicroScanner._scanInitial
package:angular_ast/src/expression/micro/scanner.dart 57:16                     NgMicroScanner.scan
package:angular_ast/src/expression/micro/lexer.dart 24:23                       NgMicroLexer.tokenize
dart:core                                                                       _SyncIterator.moveNext
package:angular_ast/src/expression/micro/parser.dart 61:20                      _RecursiveMicroAstParser.parse
package:angular_ast/src/expression/micro/parser.dart 34:7                       NgMicroParser.parse
package:angular_ast/src/expression/micro.dart 30:27                             parseMicroExpression
package:angular_ast/src/visitors/desugar_visitor.dart 195:17                    DesugarVisitor._desugarStar
package:angular_ast/src/visitors/desugar_visitor.dart 81:14                     DesugarVisitor.visitElement
package:angular_ast/src/ast/element.dart 100:20                                 _ParsedElementAst&TemplateAst&ElementAst.accept
package:angular_ast/src/visitors/desugar_visitor.dart 91:29                     DesugarVisitor._visitChildren.<fn>
dart:core                                                                       List.forEach
package:angular_ast/src/visitors/desugar_visitor.dart 90:24                     DesugarVisitor._visitChildren
package:angular_ast/src/visitors/desugar_visitor.dart 52:5                      DesugarVisitor.visitElement
package:angular_ast/src/ast/element.dart 100:20                                 _ParsedElementAst&TemplateAst&ElementAst.accept
package:angular_ast/src/visitors/desugar_visitor.dart 91:29                     DesugarVisitor._visitChildren.<fn>
dart:core                                                                       List.forEach
package:angular_ast/src/visitors/desugar_visitor.dart 90:24                     DesugarVisitor._visitChildren
package:angular_ast/src/visitors/desugar_visitor.dart 52:5                      DesugarVisitor.visitElement
package:angular_ast/src/ast/element.dart 100:20                                 _ParsedElementAst&TemplateAst&ElementAst.accept
package:angular_ast/src/parser.dart 183:25                                      NgParser.parse.<fn>
dart:core                                                                       Iterable.toList
package:angular_ast/src/parser.dart 185:12                                      NgParser.parse
package:angular_ast/angular_ast.dart 83:17                                      parse
package:angular/src/compiler/ast_directive_normalizer.dart 188:25               AstDirectiveNormalizer._parseTemplate
package:angular/src/compiler/ast_directive_normalizer.dart 147:9                AstDirectiveNormalizer._normalizeLoadedTemplate
package:angular/src/compiler/ast_directive_normalizer.dart 80:14                AstDirectiveNormalizer._normalizeTemplate
package:angular/src/compiler/ast_directive_normalizer.dart 29:12                AstDirectiveNormalizer.normalizeDirective
package:angular/src/compiler/offline_compiler.dart 92:33                        OfflineCompiler.normalizeDirectiveMetadata
dart:async                                                                      Future.wait
package:angular/src/source_gen/template_compiler/template_processor.dart 58:39  _normalizeComponent
dart:async                                                                      _completeOnAsyncReturn
package:angular/src/compiler/ast_directive_normalizer.dart                      AstDirectiveNormalizer._normalizeTemplate
dart:async                                                                      _completeOnAsyncReturn
package:pool/pool.dart                                                          Pool.withResource

**Comment by tonosama-atlacatl** [comment]
Someone alluded in Gitter that this might be related to a missing ";" in *ngFor. But, for the great majority of these issues, the template lacks any use of *ngFor.. So, if this applies to other issues, regardless, a line # or copy of the afflicting code would help tremendously.

**Comment by tonosama-atlacatl** [comment]
Thanks the wonderful people in Gitter, we managed to figure out the reasons behind this error. Basically, I have a massive "directive" export that all my components depend on...so every component has ``` directives: const[myDirectives, ...], ```, we figured out that if one component on this massive export has a INVALID_MICRO_EXPRESSION issue, all components will display the same issue, despite them not directly using such directive. My original desire still stands, as I'm sure many others might benefit from knowing the offending code/line...if its possible.

**Comment by leonsenft** [comment]
Thanks for reporting. Do you know exactly which component in your list of directives triggers this crash and would you mind sharing a reproduction of the template expression causing it?

**Comment by tonosama-atlacatl** [comment]
Not yet, hunting them down now...I will post once I find the culprit. thank you

**Comment by tonosama-atlacatl** [comment]
Yup!, confirmed... a trailing **;** in a *ngFor expression was the culprit.
```
*ngFor="let column of datagridColumns; let j = index;"
```
Unfortunately, despite cleaning up the project to not relying on the massive directive list. I wasn't able to get a more _descriptive_ error.

**Comment by leonsenft** [comment]
Well the good news is this was fixed in https://github.com/dart-lang/angular/commit/d02cbc2b4ef040efdd678b9fd0453edec275f386.

At HEAD, you should now get an error like:

```
[SEVERE]: Template parse errors:
line 1, column 18 of ... Failed parsing micro expression                                                                                           
  ╷
1 │     <div *ngFor="let item of items;">
  │                  ^^^^^^^^^^^^^^^^^^
  ╵
```

Unfortunately this fix hasn't been published yet (it was landed after our latest release 5.2.0). It will be included in our next release!

**Comment by tonosama-atlacatl** [comment]
thank you!
